pseudomonas_rnaseq_analysis/2_identify_generic_genes_pathways.py

```diff
@@ -0,0 +1,573 @@
+# source(paste0(base_dir, '/generic_expression_patterns_modules/DE_analysis.R'))
+#
+# get_DE_stats_DESeq(metadata_filename,
+#                    project_id,
+#                    processed_template_filename,
+#                    "template",
+#                    local_dir,
+#                    "real")
+
+# + magic_args="-i metadata_filename -i project_id -i base_dir -i local_dir -i num_runs -o num_sign_DEGs_simulated" language="R"
+#
+# source(paste0(base_dir,'/generic_expression_patterns_modules/DE_analysis.R'))
+#
+# num_sign_DEGs_simulated <- c()
+#
+# for (i in 0:(num_runs-1)){
+#     simulated_data_filename <- paste(
+#         local_dir,
+#         "pseudo_experiment/selected_simulated_data_",
+#         project_id,
+#         "_",
+#         i,
+#         ".txt",
+#         sep=""
+#     )
+#
+#     run_output <- get_DE_stats_DESeq(
+#         metadata_filename,
+#         project_id,
+#         simulated_data_filename,
+#         "simulated",
+#         local_dir,
+#         i
+#     )
+#     num_sign_DEGs_simulated <- c(num_sign_DEGs_simulated, run_output)
+# }
+# -
+
+# ### Rank genes
+
+analysis_type = "DE"
+template_DE_stats_filename = os.path.join(
+    local_dir, "DE_stats", f"DE_stats_template_data_{project_id}_real.txt"
+)
+template_DE_stats, simulated_DE_summary_stats = ranking.process_and_rank_genes_pathways(
+    template_DE_stats_filename,
+    local_dir,
+    num_runs,
+    project_id,
+    analysis_type,
+    col_to_rank_genes,
+    logFC_name,
+    pvalue_name,
+)
+
+# ### Gene summary table
+
+# +
+summary_gene_ranks = ranking.generate_summary_table(
+    template_DE_stats_filename,
+    template_DE_stats,
+    simulated_DE_summary_stats,
+    col_to_rank_genes,
+    local_dir,
+    "gene",
+    params,
+)
+
+summary_gene_ranks.sort_values(by="Z score", ascending=False).head()
+# -
+
+# Add gene name as column to summary dataframe
+summary_gene_ranks = ranking.add_pseudomonas_gene_name_col(summary_gene_ranks, base_dir)
+summary_gene_ranks.sort_values(by="Z score", ascending=False).head()
+
+summary_gene_ranks.sort_values(by="Percentile (simulated)", ascending=False).head()
+
+# Check if there is an NaN values, there should not be
+summary_gene_ranks.isna().any()
+
+# Create `gene_summary_filename`
+summary_gene_ranks.to_csv(gene_summary_filename, sep="\t")
+
+# ## Compare gene ranking
+#
+# We can only compare the ranking between the PAO1 RNA-seq compendium vs GAPE, where we still see good concordance as expected.
+#
+# When we look for common genes, we do this based on percentiles generated by SOPHIE for both the PAO1 and PA14 compendia to be consistent.
+
+if "pao1" in config_filename:
+    # Get generic genes identified by Crow et. al.
+    GAPE_filename = params["reference_gene_filename"]
+    ref_gene_col = params["reference_gene_name_col"]
+    ref_rank_col = params["reference_rank_col"]
+
+    figure_filename = f"gene_ranking_{col_to_rank_genes}.svg"
+
+    corr, shared_ranking = ranking.compare_gene_ranking(
```

Comment:
Is this similar to the plot you made with the other Pseudomonas compendium? Maybe you could plot them together here, or create a side by side plot somewhere to show similarity.

Reply:
So this new Pa analysis is actually supposed to be in the `core-accessory-interactome` repository. But since I needed to use SOPHIE I added it to this repo. Thank you so much for your suggestions on using submodules. I think this is what I want to be able to do. So I think that in my next PR I will pull this Pa RNA-seq analysis into my `core-accessory-interactome` repo. 

So it woudn't make too much sense to compare this analysis with the previous one if that makes sense.